engine/common/src/main/java/org/opennms/oce/engine/common/DeletingSituationHandler.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.engine.common;
+
+import java.util.Objects;
+
+import org.opennms.oce.datasource.api.SituationHandler;
+
+import org.opennms.oce.engine.api.Engine;
+
+/**
+ * A {@link SituationHandler} that handles the {@link SituationHandler#onSituationDeleted} call and triggers a
+ * {@link Engine#deleteSituation} for the given Id.
+ */
+public class DeletingSituationHandler implements SituationHandler {
```

Comment:
This class should probably belong in the `driver/main` project, since it's used exclusively by the driver.

Reply:
Ok, that was where I originally put it but it seemed like it didn't belong being the only other class there. I'll move it back.